db/src/tx.rs

```diff
@@ -263,8 +267,24 @@ impl<'conn, 'a> Tx<'conn, 'a> {
         // And evolve them forward.
         while generation.can_evolve() {
             // Evolve further.
-            let temp_id_map = self.resolve_temp_id_avs(&generation.temp_id_avs()[..])?;
+            let temp_id_map: TempIdMap = self.resolve_temp_id_avs(&generation.temp_id_avs()[..])?;
             generation = generation.evolve_one_step(&temp_id_map);
+
+            // Report each tempid that resolves via upsert.
+            for (tempid, &entid) in &temp_id_map {
```

Comment:
I'll do this, but it's not the win I think you expect.  Each `tempid` is `Rc<String>`, so we can take ownership of the `Rc` but we can't avoid the `clone` below (safely), since we might have more consumers of any individual `Rc<String>` at this point.